homework/src/index.js

```diff
@@ -30,17 +31,88 @@
+    const descriptionDetails = createAndAppend('td', descriptionTr);
+
+    const forkTr = createAndAppend('tr', table);
+    createAndAppend('td', forkTr, { 'text': 'Forks :', 'class': 'label' });
+    const forkNum = createAndAppend('td', forkTr);
+
+    const updateTr = createAndAppend('tr', table);
+    createAndAppend('td', updateTr, { 'text': 'Updated :', 'class': 'label' });
+    const updateNum = createAndAppend('td', updateTr);
+
+    url.innerText = repositoriesdata[index].name;
+    url.setAttribute('href', repositoriesdata[index].html_url);
+    if (repositoriesdata[index].description === null) {
+      descriptionName.innerHTML = '';
+      descriptionDetails.innerText = '';
+    } else {
+      descriptionName.innerText = "Description: ";
+      descriptionDetails.innerText = repositoriesdata[index].description;
+    }
+    forkNum.innerText = repositoriesdata[index].forks;
+    const updateRepo = new Date(repositoriesdata[index].updated_at);
+    updateNum.innerText = updateRepo.toLocaleString("en-US");
+  }
+
+  function renderContributors(contributorsdata, container) {
+    const contributorsDiv = createAndAppend('div', container, { 'class': 'right-div box' });
+    createAndAppend('p', contributorsDiv, { 'text': 'Contributions', 'class': 'contributions' });
+    const ul = createAndAppend('ul', contributorsDiv);
+    contributorsdata.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      li.addEventListener("click", () => { window.open(contributor.html_url); });
+      createAndAppend('img', li, { 'src': contributor.avatar_url });
+      createAndAppend('p', li, { 'text': contributor.login });
+      createAndAppend('div', li, { 'text': contributor.contributions, 'class': 'contributionNum' });
     });
   }
 
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { 'class': 'header' });
+    createAndAppend('p', header, { 'text': 'HYF Repositories' });
+    const select = createAndAppend('select', header, { 'id': 'select-list' });
+    const container = createAndAppend('div', root, { 'id': 'container' });
+
+    fetchJSON(url)
+      .then((repositories) => {
+        repositories.sort((x, y) => x.name.localeCompare(y.name));
+        repositories.forEach((item, index) => {
+          createAndAppend('option', select, { 'text': item.name, 'value': index });
+        });
+        const contributorsInfo = repositories[0].contributors_url;
+        renderRepositories(0, repositories, container);
+        select.addEventListener('change', (e) => {
+          container.innerHTML = "";
+          const index = e.target.value;
+          renderRepositories(index, repositories, container);
+          const contributorOnSelect = repositories[index].contributors_url;
+          fetchJSON(contributorOnSelect)
+            .then(contributorOnSelect => renderContributors(contributorOnSelect, container))
+            .catch(err => err.message);
```

Comment:
You are not rendering the error to the web page. You have done it correctly in line 112.